src/mono/mono/metadata/icall.c

```diff
@@ -2284,6 +2284,35 @@ typed_reference_to_object (MonoTypedRef *tref, MonoError *error)
 	HANDLE_FUNCTION_RETURN_REF (MonoObject, result);
 }
 
+gpointer
+ves_icall_System_RuntimeFieldHandle_GetFieldDataReference (MonoObjectHandle target, MonoClassField *field, MonoError *error)
+{
+	g_assert (field);
+
+	(void)mono_handle_class (target);
+
+	/* if relative, offset is from the start of target. Otherwise offset is actually an address */
+	gboolean relative = TRUE;
+	intptr_t offset = 0;
+	if (G_LIKELY (!m_field_is_from_update (field))) {
+		offset = m_field_get_offset (field);
+	} else {
+		/* This field was added by a metadata-update to an exsiting type.
+			* Since it's store outside the object, offset is an absolute address
+			*/
+		relative = FALSE;
+		uint32_t token = mono_metadata_make_token (MONO_TABLE_FIELD, mono_metadata_update_get_field_idx (field));
+		offset = (intptr_t) mono_metadata_update_added_field_ldflda (MONO_HANDLE_RAW (target), field->type, token, error);
+		mono_error_assert_ok (error);
+	}
+	(void)mono_class_from_mono_type_internal (field->type);
```

Comment:
why? 

if you need the field type to be fully inited, this isn't enough.  you would probably need `mono_class_init_internal (mono_class_from_mono_type_internal (field->type))`

Reply:
Again, copy from previous implementation.

Reply:
Will remove.